scripts/yaml_check.py

```diff
@@ -0,0 +1,41 @@
+import yaml
+import sys
+
+def tiered_extract(db):
+  out = []
+  for _, entry in db.items():
+    for _, values in entry.items():
+      for name, decl in values.items():
+        if decl is not None and decl != '' and '/' not in decl:
+          out.append((name, decl))
+  return out
```

Comment:
You don't need those `_, ...`, you can replace `.items()` by `.values()`. You would get:
```python
  for entry in db.values():
    for values in entry.values():
      for name, decl in values.items():
        if decl and '/' not in decl:
          out.append((name, decl))
  return out
```
Of course you can also go crazy and use:
```python
  return [(name, decl) for entry in db.values() for values in entry.values() for name, decl in values.items() if decl and '/' not in decl]
```
Note that iterated list comprehension syntax is designed so that it reads in the same order as your nested loops, except for the value which comes first. Choosing between those two options is mostly a matter of taste.

What is more important is documentation, using docstrings and type annotations. Something like:
```python
def tiered_extract(db: Dict[str, Dict[str, Dict[str, Optional[str]]]]) -> List[Tuple[str, str]]:
  """From a three-level deep nested dictionary, return a list of (key, values)  
  of the deepest level."""
  return ...
```
after adding in the imports `from typing import Dict, Optional, Tuple, List`.